docs/development.rst

```diff
@@ -0,0 +1,198 @@
+.. _Magic-Folder Development:
+
+Magic-Folder Development
+========================
+
+Process Overview
+----------------
+
+The high-level view of the Magic-Folder development process looks like this:
+
+#. A shortcoming, defect, or new area of functionality is identified relating to the software.
+   For example,
+   perhaps files with names matching a certain pattern are never uploaded to the grid
+   or a developer decides to add a history browsing user experience to the software.
+#. `An issue is filed`_ describing the shortcoming, defect, or new area of functionality.
+   The issue describes how the software will be different than it is now and why this is desirable.
+#. Development work proceeds in a branch (typically created from a recent master branch revision).
+   Prior to this development or in parallel with it,
+   communication within the team helps identify and clear roadblocks early.
+#. When development has progressed sufficiently,
+   the branch is `proposed for review`_.
+#. One or more other developers `review the changes`_.
+#. The previous two steps may repeat a number of times until the changes pass review.
+#. Upon passing review,
+   the changes are `merged into the master branch`_.
+
+
+.. _An issue is filed:
+
+Filing Issues
+-------------
+
+The project uses `GitHub issues <https://github.com/leastauthority/magic-folder/issues>`_ for issue tracking.
+
+.. _proposed for review:
+
+Proposing Changes
+-----------------
+
+To propose a change,
+create a `GitHub Pull Request <https://github.com/leastauthority/magic-folder/pulls>`_ against master.
+
+Changes should:
+
+  * have a corresponding ticket.
+  * be in a branch named ``<ticket number>.<some short descriptive text>``.
+  * have appropriate developer-facing and user-facing documentation.
+  * have full line and branch coverage provided by the test suite.
+  * be implemented in a way that is not unnecessarily difficult to understand and maintain.
+  * satisfy all of the mechnical checks performed by the continuous integration system.
+
+News Fragments
+~~~~~~~~~~~~~~
+
+One of the mechnical checks performed by continuous integration against proposed changes is the existence of a "news fragment".
+News fragments are assembled at release time by `towncrier <https://pypi.org/project/towncrier/>`_ to contribute to the release announcement.
+News fragments are meant to be user-facing and should have a consistent style.
+News fragments can use a inline formatting directives from reStructuredText.
+
+magic-folder's news fragment style is adapted from `the style guidelines from the Twisted project>`_.
```

Comment:
That trailing `>` would be a problem when rendering this document as HTML, as in:

https://github.com/LeastAuthority/magic-folder/blob/164.dev-docs-etc/docs/development.rst#news-fragments